modules/core/src/main/scala/fs2/kafka/ProducerSettings.scala

```diff
@@ -384,62 +383,51 @@ object ProducerSettings {
         }
     )
 
-  /**
-    * Creates a new [[ProducerSettings]] instance using the
-    * specified [[Serializer]]s for the key and value.
-    */
   def apply[F[_], K, V](
     keySerializer: Serializer[F, K],
     valueSerializer: Serializer[F, V]
   )(implicit F: Sync[F]): ProducerSettings[F, K, V] =
-    create(F.pure(keySerializer), F.pure(valueSerializer))
+    create(
+      keySerializer = F.pure(keySerializer),
+      valueSerializer = F.pure(valueSerializer)
+    )
 
-  /**
-    * Creates a new [[ProducerSettings]] instance using the
-    * specified [[Serializer]]s for the key and value, in
-    * the case where the key serializer is wrapped in a
-    * creation effect.
-    */
   def apply[F[_], K, V](
-    keySerializer: F[Serializer[F, K]],
+    keySerializer: Serializer.Record[F, K],
     valueSerializer: Serializer[F, V]
   )(implicit F: Sync[F]): ProducerSettings[F, K, V] =
-    create(keySerializer, F.pure(valueSerializer))
+    create(
+      keySerializer = keySerializer.forKey,
+      valueSerializer = F.pure(valueSerializer)
+    )
 
-  /**
-    * Creates a new [[ProducerSettings]] instance using the
-    * specified [[Serializer]]s for the key and value, in
-    * the case where the value serializer is wrapped in a
-    * creation effect.
-    */
   def apply[F[_], K, V](
     keySerializer: Serializer[F, K],
-    valueSerializer: F[Serializer[F, V]]
+    valueSerializer: Serializer.Record[F, V]
   )(implicit F: Sync[F]): ProducerSettings[F, K, V] =
-    create(F.pure(keySerializer), valueSerializer)
+    create(
+      keySerializer = F.pure(keySerializer),
+      valueSerializer = valueSerializer.forValue
+    )
 
-  /**
-    * Creates a new [[ProducerSettings]] instance using the
-    * specified [[Serializer]]s for the key and value, in
-    * the case where both the key and value serializers
-    * are wrapped in a creation effect.
-    */
   def apply[F[_], K, V](
-    keySerializer: F[Serializer[F, K]],
-    valueSerializer: F[Serializer[F, V]]
+    keySerializer: Serializer.Record[F, K],
+    valueSerializer: Serializer.Record[F, V]
   )(implicit F: Sync[F]): ProducerSettings[F, K, V] =
-    create(keySerializer, valueSerializer)
+    create(
```

Comment:
I find this quite confusing to take two `Serializer.Record` but only use the key / value side of it

Reply:
It's valid though, since `Serializer.Record` provides two serializers.

For example:

```scala
val serializer = avroSerializer[String].using(avroSettings)

ProducerSettings(
  keySerializer = serializer,
  valueSerializer = serializer
)
```

would use both parts, or the following:

```scala
ProducerSettings(
  keySerializer = avroSerializer[String].using(avroSettings),
  valueSerializer = avroSerializer[Person].using(avroSettings)
)
```

where you would use Avro for both serializers, but only the `forKey` part for `String` and the `forValue` part for `Person`.